tests/e2e/tests/pilot/routing_test.go

```diff
@@ -858,7 +858,7 @@ func TestSidecarScope(t *testing.T) {
 			testName:  "ns1: bookinfo.com:9999 reachable via egress TCP listener 7.7.7.7:23145",
 			reqURL:    "http://7.7.7.7:23145/a",
 			host:      "bookinfo.com:9999",
-			reachable: true,
+			reachable: false, // TODO: This needs to be changed back to true once we handle "HTTP_PROXY" protocol in sidecar.
```

Comment:
I am confused why this changed

Reply:
Because the sidecar config here https://github.com/istio/istio/blob/master/tests/e2e/tests/pilot/testdata/networking/v1alpha3/sidecar-scope-ns1-ns2.yaml only has port 23145 and bookinfo.com does not listen on that port https://github.com/istio/istio/blob/master/tests/e2e/tests/pilot/testdata/networking/v1alpha3/service-entry-http-scope-public.yaml#L9
Earlier for wildcard we just skipped port matching. To support this usecase, we will have to define the port in egress listener as "PROXY_PORT" (in protocol) and if it is specified, we treat it as regular egress http proxy and stop filtering. Then it will work. I plan to followup with that change once we are good with this.